pkg/typesystem/typesystem.go

```diff
@@ -568,8 +594,15 @@ func Validate(model *openfgapb.AuthorizationModel) error {
 		return ErrDuplicateTypes
 	}
 
-	if err := validateRelationRewrites(model); err != nil {
-		return err
+	// Validate the userset rewrites
+	typesys := New(model)
```

Comment:
Does it feel weird to construct a typesystem inside of the method that should be validating a model before a typesystem is constructed? I'm leaning towards this method operating on the primitives and then the typesystem can operate on the assumption that it's constructed over a valid model.

Reply:
That is a good point. Let me try some stuff.

Reply:
Refactored as discussed offline